keyboards: Add support for Sofle Choc

t-8ch opened the pull request:
This PR is to review and test my changes for the Sofle Choc upstreaming process.

@brianlow @duese  are you fine with me squashing all your work into this commit together?
There is authorship attribution in the commit message.

t-8ch committed:
keyboards: Add support for Sofle Choc

Co-developed-by: Brian Low <github@brianlow.com>
Co-developed-by: duese <duese@users.noreply.github.com>

brianlow reviewed (approved):
LGTM. Squash ok with me. How do plan to get this into PR? Force push the choc2 branch?

t-8ch commented:
> Force push the choc2 branch?

Yes.

But before that somebody needs to test it on a physical device.

duese commented:
I can test it tonight.

duese commented:
Hey. Finally had time to test this PR. Squashing is fine for me too.
Unfortunately something isn't right. After flashing the two halves, the OLED display on the left half shows some crippled artefacts as shown in the attached photo.
![signal-2023-08-08-072629_002](https://github.com/brianlow/qmk_firmware/assets/1257829/96e4418a-4982-4504-83d8-4cbb65af547a)

My Macbook is up to date, but while flashing I get the following output:
```
> qmk flash -kb sofle_choc -km default
Ψ Compiling keymap with gmake --jobs=1 sofle_choc:default:flash


QMK Firmware 0.21.6
Making sofle_choc with keymap default and target flash

avr-gcc (Homebrew AVR GCC 8.5.0_1) 8.5.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Size before:
   text	  data	   bss	   dec	   hex	filename
      0	 24764	     0	 24764	  60bc	sofle_choc_default.hex

Copying sofle_choc_default.hex to qmk_firmware folder                                               [OK]
Checking file size of sofle_choc_default.hex                                                        [OK]
 * The firmware size is fine - 24764/28672 (86%, 3908 bytes free)
Flashing for bootloader: caterina
Waiting for USB serial port - reset your controller now (Ctrl+C to cancel).....
Device /dev/tty.usbmodem14201 has appeared; assuming it is the controller.
Waiting for /dev/tty.usbmodem14201 to become writable.
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:404]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:405]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:433]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:434]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:466]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:467]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:488]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:489]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:552]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:553]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:572]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:573]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:591]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:592]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:622]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:623]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:676]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:677]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:760]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:762]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:771]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:773]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:783]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:785]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:796]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:798]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:808]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:809]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:831]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:833]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:855]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:857]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:1433]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:1434]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:1455]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:1456]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:1469]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:1470]
avrdude warning: mosi is deprecated, will be removed in v8.0, use sdo [/usr/local/etc/avrdude.conf:1483]
avrdude warning: miso is deprecated, will be removed in v8.0, use sdi [/usr/local/etc/avrdude.conf:1484]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:1708]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:3199]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:3388]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:3577]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:3769]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:3961]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:4154]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:4340]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:4664]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:4883]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:5083]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:5439]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:5651]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:5905]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:6139]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:7104]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:7301]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:7498]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:7755]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:7982]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:8209]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:8438]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:8635]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:8821]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:9008]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:9196]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:9384]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:9725]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:9923]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:10107]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:10390]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:10574]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:10757]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:10943]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:11133]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:11336]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:11778]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:11971]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:12164]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:12480]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:12663]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:12854]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13045]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13249]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13451]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13452]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13638]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13639]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13827]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:13828]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14015]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14016]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14203]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14204]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14387]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14609]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:14829]
avrdude warning: has_pdi is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15108]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15236]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15325]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15425]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15585]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15643]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15755]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15887]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:15987]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:16264]
avrdude warning: is_avr32 is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:16265]
avrdude warning: has_debugwire is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:16289]
avrdude warning: has_tpi is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:16487]
avrdude warning: has_jtag is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:16644]
avrdude warning: has_updi is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:16709]
avrdude warning: has_updi is deprecated, will be removed in v8.0, use prog_modes [/usr/local/etc/avrdude.conf:18001]
avrdude: AVR device initialized and ready to accept instructions
avrdude: device signature = 0x1e9587 (probably m32u4)
avrdude: Note: flash memory has been specified, an erase cycle will be performed.
         To disable this feature, specify the -D option.
avrdude: erasing chip

avrdude: processing -U flash:w:.build/sofle_choc_default.hex:i
avrdude: reading input file .build/sofle_choc_default.hex for flash
         with 24764 bytes in 1 section within [0, 0x60bb]
         using 194 pages and 68 pad bytes
avrdude: writing 24764 bytes flash ...
Writing | ################################################## | 100% 1.90 s 
avrdude: 24764 bytes of flash written
avrdude: verifying flash memory against .build/sofle_choc_default.hex
Reading | ################################################## | 100% 0.20 s 
avrdude: 24764 bytes of flash verified

avrdude done.  Thank you.
```
I'm not sure whether the warnings can be ignored.

`qmk doctor` does not complain:
```
> qmk doctor
Ψ QMK Doctor is checking your environment.
Ψ CLI version: 1.1.2
Ψ QMK home: /Users/duese/dev/qmk
Ψ Detected macOS 13.5 (Intel).
Ψ Git branch: sofle_choc
Ψ Repo version: 0.21.6
Ψ - Latest sofle_choc: 2023-07-30 17:24:39 +0200 (115b82511e) -- keyboards: Add support for Sofle Choc
Ψ - Latest upstream/master: 2023-08-07 03:13:20 +1000 (d52bafade3) -- Fix `binepad/bn009` and `durgod/dgk6x/hades_ansi` keymaps (#21701)
Ψ - Latest upstream/develop: 2023-08-06 17:13:52 +0000 (2e67945033) -- Merge remote-tracking branch 'origin/master' into develop
Ψ - Common ancestor with upstream/master: 2023-07-30 00:22:39 -0400 (14e14e9ab8) -- Correct "less than" to "up to" in squeezing_avr?id=layers (#21639)
Ψ - Common ancestor with upstream/develop: 2023-07-30 04:23:13 +0000 (fa4d51dab7) -- Merge remote-tracking branch 'origin/master' into develop
Ψ CLI installed in virtualenv.
Ψ All dependencies are installed.
Ψ Found arm-none-eabi-gcc version 8.3.1
Ψ Found avr-gcc version 8.5.0
Ψ Found avrdude version 7.2
Ψ Found dfu-programmer version 1.1.0
Ψ Found dfu-util version 0.11
Ψ Submodules are up to date.
Ψ Submodule status:
Ψ - lib/chibios: 2023-04-15 13:48:04 +0000 --  (11edb1610)
Ψ - lib/chibios-contrib: 2023-07-17 11:39:05 +0200 --  (da78eb37)
Ψ - lib/googletest: 2021-06-11 06:37:43 -0700 --  (e2239ee6)
Ψ - lib/lufa: 2022-08-26 12:09:55 +1000 --  (549b97320)
Ψ - lib/vusb: 2022-06-13 09:18:17 +1000 --  (819dbc1)
Ψ - lib/printf: 2022-06-29 23:59:58 +0300 --  (c2e3b4e)
Ψ - lib/pico-sdk: 2023-02-12 20:19:37 +0100 --  (a3398d8)
Ψ - lib/lvgl: 2022-04-11 04:44:53 -0600 --  (e19410f8)
Ψ QMK is ready to go
```

Maybe I can investigate later.

t-8ch commented:
The warnings during flashing seem unrelated.

I'll take a look at the display, did it work with the original PR?

Are keypresses, LEDs working?

duese commented:
> Are keypresses, LEDs working?

Regular keypresses without modifier keys are working normal. Shift modifier works too. It seems the layer shifting modifiers don't work as shown on in the default layout depicted in the [build guide](https://josefadamcik.github.io/SofleKeyboard/build_guide_choc.html). E.g. when pressing any layer modifier in combination with "`asdfg`" I get "`åß∂ƒ©`" instead.

I can only see the default red pulsing. Due to the disfunctional layer modifiers I can't switch the animation and/or LED color.

t-8ch commented:
The layers being gone is intentional, as the maintainers requested a simpler keymap by default.
I guess this means the LEDs should default to something more conservative.

duese commented:
> The layers being gone is intentional, as the maintainers requested a simpler keymap by default.

Do you really think this is what Drashna means with "pristine/simple"? Even the corne keyboard has layers in [its default keymap](https://github.com/qmk/qmk_firmware/blob/master/keyboards/crkbd/keymaps/default/keymap.c). And the default map has Drashna in its copyright message.

But to get this board merged I would be ok with that, but then I would appreciate it if you move the previous default map to something like "old default", "brians default" or "creators default". What good is it having a RGB board if you can't change anything RBG related?

duese commented:
If one can't change the animation you should then also comment out all other animations except for the one used in `config.h`

t-8ch commented:
> Do you really think this is what Drashna means with "pristine/simple"?

This was my impression, yes.

> I would appreciate it if you move the previous default map to something like "old default", "brians default" or "creators default"

I would prefer to keep this out of the initial PR. Let's focus on the minimal amount of stuff to make it useful and get it merged.
Currently there is already an issue with reviewer capacity additional work needed.
Any additional code will only make the process slower and more work for everybody.

The original layout is a nice example but it's much easier to add it after the initial merge.

> also comment out all other animations except for the one used in config.h

I'll delete them completely for the first merge.
They are not lost and can be readded as part of brians keymap after the board support has landed.

duese commented:
Agreed.

> I'll delete them completely for the first merge.

Do that and I'll approve.

duese commented:
And did you look into the the crippeled OLED display?

t-8ch commented:
I didn't have time yet to look at the OLED. Will do so later.
Just to be sure: The OLED does work correctly on the original upstreaming PR?

duese commented:
> Just to be sure: The OLED does work correctly on the original upstreaming PR?

What upstreaming PR do mean? I checked out your PR without any modifications, compiled and flashed it to my Sofle Choc.

t-8ch commented:
https://github.com/qmk/qmk_firmware/pull/16736

duese commented:
> https://github.com/qmk/qmk_firmware/pull/16736

Had to merge upstream/develop (cf5626e0246b66879b03a18115a19a566e839f48) into choc2 to get that PR to compile but then I could flash it and it worked.
No issues with the OLED display there.

t-8ch commented:
@duese Thanks for testing! Care to try again?

duese commented:
@t-8ch Flashed the current head. All basic keypresses work as expected. The LEDs glow constantly in red - no pulsing anymore. I assume this is intentional. Also both OLEDs now show the QMK logo. No status is shown anymore. I guess this is also intentional as there are no layer modifiers anymore.

Thanks for the work!

t-8ch commented:
Thanks for the review!

@brianlow Could you also give your sign-off?
I'd like to push this to your `choc2` branch and update the original PR.

brianlow commented:
Go for it! Thanks

dunk2k committed:
[Keyboard] Update usb.* of `dactyl_lightcycle` (#21462)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

dunk2k committed:
[Keyboard] add laneware `raindrop` keyboard (#21277)

Co-authored-by: Sergey Vlasov <sigprof@gmail.com>
Co-authored-by: Ryan <fauxpark@gmail.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

gskygithub committed:
[Keyboard] Add Projectd/65/Projectd_65_ansi (#21520)

Co-authored-by: Ryan <fauxpark@gmail.com>
Co-authored-by: gksygithub <106651989+gksygithub@users.noreply.github.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

puffnfresh committed:
[Keymap] Add Via keymap for CXT Studio (#21675)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

spbgzh committed:
[Keyboard] fix json for nemui65 (#21629)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

dunk2k committed:
[Keyboard] Update USB settings for dactyl_maximus (#21666)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

jonylee1986 committed:
[Keyboard] Add mk47 keyboard (#19797)

Co-authored-by: Drashna Jaelre <drashna@live.com>
Co-authored-by: Ryan <fauxpark@gmail.com>
Co-authored-by: jack <0x6a73@protonmail.com>
Co-authored-by: Joel Challis <git@zvecr.com>
Co-authored-by: Joy <chang.li@westberrytech.com>
Co-authored-by: Joy Lee <joylee.lc@foxmail.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

dunk2k committed:
correct keymap file name for tkl_f13_jis layout (#21631)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

HorrorTroll committed:
[Keyboard] Refactor Caticorn PCB (#21644)

alvicstep committed:
Kintwin controller for kinesis keyboard, split layout (#21614)

Co-authored-by: Ryan <fauxpark@gmail.com>

fauxpark committed:
Fix `keyboard` field for some JSON keymaps (#21752)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

mechlovin committed:
[Keyboard] Add Shostudio Arc (#21733)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

rpbaptist committed:
[Keymap] Update rpbaptist crkbd (#21674)

Co-authored-by: Drashna Jaelre <drashna@live.com>
Co-authored-by: Richard Baptist <richard@remote.com>

andrebrait committed:
[Keymap] Update andrebrait keymap for GMMK Pro (#21668)

janat-t committed:
[Keymap] submitting janat-t planck keymap (#21619)

Co-authored-by: Janat Taerakul <jtaerakul@indeed.com>

edwardslau committed:
[keyboard] Graystudio Think6.5 V3 Support Revised (#21560)

Co-authored-by: jack <0x6a73@protonmail.com>
Co-authored-by: Duncan Sutherland <dunk2k_2000@hotmail.com>
Co-authored-by: Ryan <fauxpark@gmail.com>

zvecr committed:
Remove invalid config option from kikoslab/kl90 (#21708)

dlip committed:
[Keyboard] Add Chouchou keyboard (#21699)

fauxpark committed:
OLED: driver naming cleanups (#21710)

dunk2k committed:
Update usb information for dactyl_cc (#21612)

zvecr committed:
Merge remote-tracking branch 'origin/master' into develop

fauxpark committed:
rgblight: driver selection cleanups (#21558)

willoucom committed:
[Keyboard] add willoucom/keypad (#21714)

joe-scotto committed:
Add ScottoErgo Keyboard (#20944)

Co-authored-by: Drashna Jaelre <drashna@live.com>
Co-authored-by: Ryan <fauxpark@gmail.com>

joe-scotto committed:
Add Scotto34 Keyboard (#20943)

Co-authored-by: Albert Y <76888457+filterpaper@users.noreply.github.com>
Co-authored-by: Drashna Jaelre <drashna@live.com>
Co-authored-by: Ryan <fauxpark@gmail.com>

joe-scotto committed:
Add ScottoInvader Keyboard (#20941)

Co-authored-by: Ryan <fauxpark@gmail.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

fauxpark committed:
coarse/cordillera: fix backlight config (#21761)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

Commit:
The PoweredByPorridge layout for the Keyboardio Atreus (#21737)

Co-authored-by: jack <0x6a73@protonmail.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

squigglybob committed:
[Keymap] add squigglybob splitkb kyria rev2 keymap (#21751)

Co-authored-by: Duncan Sutherland <dunk2k_2000@hotmail.com>
Co-authored-by: jack <0x6a73@protonmail.com>

drashna committed:
[Bug] Fix compilation error for APA on ChibiOS (#21773)

studiokestra committed:
Add Galatea rev3 keyboard (#21758)

Pangorin committed:
Adding support for TAN67 PCB (#21003)

Co-authored-by: Drashna Jaelre <drashna@live.com>
Co-authored-by: Ryan <fauxpark@gmail.com>
Co-authored-by: Duncan Sutherland <dunk2k_2000@hotmail.com>

lizhenmingdirk committed:
[Keyboard] Add Jelly Evolv (#21407)

Co-authored-by: jack <0x6a73@protonmail.com>

duilchoi committed:
[Keyboard] Add linworks favepada (#20794)

Co-authored-by: Drashna Jaelre <drashna@live.com>
Co-authored-by: Duncan Sutherland <dunk2k_2000@hotmail.com>
Co-authored-by: jack <0x6a73@protonmail.com>

zvecr committed:
Initial support for ALT v2 (#21754)

zvecr committed:
Initial support for CTRL v2 (#21755)

zvecr committed:
Initial support for SHIFT v2 (#21756)

zvecr committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Align SENSE75 with recent Drop additions (#21757)

zvecr committed:
Fix drop/shift/v2 compilation (#21800)

controller-works committed:
[Keyboard] Add city42 (#21727)

Co-authored-by: Joel Challis <git@zvecr.com>
Co-authored-by: Duncan Sutherland <dunk2k_2000@hotmail.com>

lesshonor committed:
fix: restore indicators to jellybean_raindrops (#21792)

After #21169, rgb_matrix_indicators() was limited to running on the
final render iteration. Since the jellybean_raindrops animation
immediately returns false after updating a single LED, the iteration
count no longer ends up high enough to render non-advanced indicators.

This change also brings jellybean_raindrops more in line with raindrops.

zvecr committed:
Simplfy RGB/LED matrix effect logic (#21703)

fauxpark committed:
RGB Matrix: driver naming cleanups (#21594)

fauxpark committed:
Fix keyboards with old RGB driver names (#21815)

filterpaper committed:
[Doc] Improve magic keycode section (#21787)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Fix keyboards with old RGB driver names (#21817)

sigprof committed:
keychron/c2_pro/ansi/white: Fix column 19 in the custom matrix (#21805)

Although `keychron/c2_pro/ansi/rgb` and `keychron/c2_pro/ansi/white` use
the same custom matrix code, the matrix layouts are slightly different;
in particular, only the `keychron/c2_pro/ansi/white` board actually uses
column 19.  However, the handling of column 19 in the custom matrix code
was broken, therefore that column did not work.

Looks like the custom matrix code assumes that `SHIFT_COL_END` refers to
the last column connected to the shift register, and not to the column
past that; so the value of `SHIFT_COL_END` needs to be changed from 19
to 18 (columns 11...18 are connected to the shift register, and column
19 is connected to the C14 pin).

Also the code which was determining `SIZE_T` and `UNSELECT_ALL_COL` had
an off-by-one bug when counting the required number of bits (again due
to the confusion on the `SHIFT_COL_END` meaning); this had been fixed
too (the actual behavior of that part of the code did not change,
because both the old and the new version select the 8 bit variant).

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Fix infinite recursion in lxxt encoder logic (#21819)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

dependabot[bot] committed:
Bump tj-actions/changed-files from 37 to 38 (#21822)

Bumps [tj-actions/changed-files](https://github.com/tj-actions/changed-files) from 37 to 38.
- [Release notes](https://github.com/tj-actions/changed-files/releases)
- [Changelog](https://github.com/tj-actions/changed-files/blob/main/HISTORY.md)
- [Commits](https://github.com/tj-actions/changed-files/compare/v37...v38)

---
updated-dependencies:
- dependency-name: tj-actions/changed-files
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

t-8ch committed:
Generalize AVR -Werror=array-bounds workaround (#21798)

GCC 13 needs the same workaround as GCC 12.
To avoid having to maintain an ever-growing list of broken versions
apply the workaround wherever it can be used.

If at some point a fixed version of GCC is released the workaround can
be disabled for those fixed versions.

See #17064

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

fauxpark committed:
Move RGBLight animations to data driven (#21635)

* Move RGBLight animations to data driven, 0-9

* Move RGBLight animations to data driven, A

* Move RGBLight animations to data driven, B

* Move RGBLight animations to data driven, C

* Move RGBLight animations to data driven, D

* Move RGBLight animations to data driven, E

* Move RGBLight animations to data driven, F

* Move RGBLight animations to data driven, G

* Move RGBLight animations to data driven, H

* Move RGBLight animations to data driven, handwired

* Move RGBLight animations to data driven, I

* Move RGBLight animations to data driven, J

* Move RGBLight animations to data driven, K

* Move RGBLight animations to data driven, L

* Move RGBLight animations to data driven, M

* Move RGBLight animations to data driven, N

* Move RGBLight animations to data driven, O

* Move RGBLight animations to data driven, P

* Move RGBLight animations to data driven, Q

* Move RGBLight animations to data driven, R

* Move RGBLight animations to data driven, S

* Move RGBLight animations to data driven, T

* Move RGBLight animations to data driven, U

* Move RGBLight animations to data driven, V

* Move RGBLight animations to data driven, W

* Move RGBLight animations to data driven, X

* Move RGBLight animations to data driven, Y

* Move RGBLight animations to data driven, Z

* Fix incorrect placement

* Fix build failures and mismatches

patrickxiong committed:
converter/ibm_terminal: Fix F0_BREAK state transition (#21824)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

zvecr committed:
Remove `led_matrix.hue_steps` and `led_matrix.sat_steps` from schema (#21827)

HiryKun committed:
Add "apm32-dfu" in keyboard.jsonschema (#21842)

andysmithfal committed:
Corrected duration example, should be number not string (#21839)

qmk-bot committed:
Merge remote-tracking branch 'origin/master' into develop

fauxpark committed:
Rework info.json reference (#21324)

* Rework info.json reference

* Apply suggestions from code review

Co-authored-by: Sergey Vlasov <sigprof@gmail.com>

* Update backlight.pins description

* Add rgblight.driver and lowercase other driver names

* Remove `split.main` and `split.matrix_grid` for now as they are not working yet

---------

Co-authored-by: Sergey Vlasov <sigprof@gmail.com>

drashna committed:
Revert  changes to ChibiOS Suspend Code (#21830)

* Partially revert #19780

* Finish

* Get teensy 3.5/3.6 board files too

* fix lint issue

* Revert "[Bug] Restore usb suspend wakeup delay (#21676)"

This reverts commit e8e989fd7ad7c10e725e50ae8b0a4426e09f7f30.

* Apply suggestions from code review

Co-authored-by: Joel Challis <git@zvecr.com>

---------

Co-authored-by: Joel Challis <git@zvecr.com>

zvecr committed:
Add *_MATRIX_LED_COUNT generation (#21110)

fauxpark committed:
Unicode, Unicodemap and UCIS refactor (#21659)

fauxpark committed:
Clean up Unicode API usage in user keymaps (#21849)

tzarc committed:
Breaking changes changelog, 2023q3 version (#21848)

Co-authored-by: James Young <18669334+noroadsleft@users.noreply.github.com>
Co-authored-by: Ryan <fauxpark@gmail.com>
Co-authored-by: honorless <86894501+lesshonor@users.noreply.github.com>

tzarc committed:
Remove develop notice.

tzarc committed:
Merge branch 'develop'
